lib/trento.pm

```diff
@@ -0,0 +1,300 @@
+use utils qw(script_retry);
+use Carp qw(croak);
+use Exporter 'import';
+
+our @EXPORT = qw(
+  TRENTO_AZ_PREFIX
+  get_resource_group
+  get_vm_name
+  get_acr_name
+  az_get_vm_ip
+  VM_USER
+  SSH_KEY
+  cypress_install_container
+  PODMAN_PULL_LOG
+  CYPRESS_LOG_DIR
+  cypress_exec
+  cypress_test_exec
+  cypress_log_upload
+);
+
+=head1 SYNOPSIS
+Package with common methods and default values for Trento tests
+
+=cut
+
+use constant TRENTO_AZ_PREFIX => 'openqa-trento';
+# Parameter 'registry_name' must conform to the following pattern: '^[a-zA-Z0-9]*$'.
+use constant TRENTO_AZ_ACR_PREFIX => 'openqatrentoacr';
+
+
+=head2 get_resource_group
+Return a string to be used as cloud resource group.
+It contains the JobId
+
+=cut
+sub get_resource_group {
+    my $job_id = get_current_job_id();
+    return TRENTO_AZ_PREFIX . "-rg-$job_id";
+}
+
+
+=head2 get_vm_name
+Return a string to be used as cloud VM name.
+It contains the JobId
+
+=cut
+sub get_vm_name {
+    my $job_id = get_current_job_id();
+    return TRENTO_AZ_PREFIX . "-vm-$job_id";
+}
+
+
+=head2 get_acr_name
+Return a string to be used as cloud ACR name.
+It contains the JobId
+
+=cut
+sub get_acr_name {
+    return TRENTO_AZ_ACR_PREFIX . get_current_job_id();
+}
+
+
+=head2 az_get_vm_ip
+Return the running VM IP
+
+=cut
+sub az_get_vm_ip {
+    my $az_cmd = 'az vm show -d ' .
+      '-g ' . get_resource_group() . ' ' .
+      '-n ' . get_vm_name() . ' ' .
+      '--query "publicIps" ' .
+      '-o tsv';
+    return script_output($az_cmd, 180);
+}
+
+
+=head2 az_delete_group
+Delere resource group and so all its content
+
+=cut
+sub az_delete_group {
+    script_run('echo "Delete all resources"');
+    my $az_cmd = 'az group delete ' .
+      '--resource-group ' . get_resource_group() .
+      ' --yes';
+    script_retry($az_cmd, timeout => 600, retry => 5, delay => 60);
+}
+
+
+use constant VM_USER => 'cloudadmin';
+use constant SSH_KEY => '/root/.ssh/id_rsa';
+
+
+=head2 az_vm_ssh_cmd
+Compose ssh command for remote ssh execution on the VM machine
+The function optionally accept the vm ip.
+If it is not provided it is calculated on the fly with an az query,
+take care that is a time consuming az query.
+=cut
+sub az_vm_ssh_cmd {
+    my ($self, $cmd_arg, $vm_ip_arg) = @_;
+    my $vm_ip;
+    record_info('REMOTE', "cmd:$cmd_arg");
+    if (defined $vm_ip_arg) {
+        $vm_ip = $vm_ip_arg;
+    }
+    else {
+        $vm_ip = az_get_vm_ip();
+    }
+    return 'ssh' .
+      ' -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR' .
+      ' -i ' . SSH_KEY . ' ' .
+      VM_USER . '@' . $vm_ip .
+      ' -- ' . $cmd_arg;
+}
+
+
+=head2 k8s_logs
+Get all relevan tinfo out from the cluster
+=cut
+sub k8s_logs {
+    my $self = shift;
+    my (@pods_list) = @_;
+    my $machine_ip = az_get_vm_ip;
+
+    if ($machine_ip ne "") {
+        my $kubectl_pods = script_output($self->az_vm_ssh_cmd('kubectl get pods', $machine_ip), 180);
```

Comment:
code to be improved as it could result in teardown to fails http://1c242.qa.suse.de/tests/383#step/deploy_trento/28  fails as bash: kubectl: command not found  due to fact that teardown code try to collect cluster logs but the cluster is not there yet.